demux/demux_lavf.c

```diff
@@ -941,8 +955,11 @@ static void nested_io_close(struct AVFormatContext *s, AVIOContext *pb)
         }
     }
 
-
+#if HAVE_IO_CLOSE2
+    return priv->default_io_close2(s, pb);
+#else
     priv->default_io_close(s, pb);
+#endif
 }
```

Comment:
the ifdefs here aren't exactly pretty but I don't think there's a better solution.

Reply:
Yeah. There's some overly clever things you could imagine doing with a `#define` for the return type and using the same name for the function in all cases, but that seems potentially more confusing than helpful.